src/system/Button/Button.js

```diff
@@ -3,44 +3,60 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { useCallback } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { Button as ThemeButton } from 'theme-ui';
 
-const Button = React.forwardRef( ( { sx, ...props }, forwardRef ) => (
-	<ThemeButton
-		sx={ {
-			verticalAlign: 'middle',
-			display: 'inline-flex',
-			alignItems: 'center',
-			justifyContent: 'center',
-			height: '36px',
-			py: 0,
-			textDecoration: 'none',
-			'&:hover': {
+const Button = React.forwardRef( ( { disabled, onClick, sx, ...props }, forwardRef ) => {
+	const handleOnClick = useCallback(
+		event => {
+			if ( disabled ) {
+				event.preventDefault();
+			} else if ( onClick ) {
+				onClick( event );
+			}
+		},
+		[ disabled, onClick ]
+	);
```

Comment:
```suggestion
	const handleOnClick = useCallback(
		event => {
			if ( disabled ) {
			  return event.preventDefault();
			}
			
			if ( onClick ) {
			  return onClick( event );
			}
		},
		[ disabled, onClick ]
	);
```

Could this be simpler? I think if it's not disabled, you still want to call the `onClick`, right? So, in this case could not be a `else if`.